src/main/scala/org/apache/spark/ExposeJobListener.scala

```diff
@@ -0,0 +1,9 @@
+package org.apache.spark
+import org.apache.spark.ui.jobs.JobProgressListener
+
+object ExposeJobListener {
+  def getjobListener(sparkContext: SparkContext): JobProgressListener = {
+    sparkContext.jobProgressListener
+  }
+}
+
```

Comment:
I wasn't aware of this little trick for exposing privately held attributes. Learned something new here :-)